### ISSUE TITLE
Sync failed: No bookmarks found for the provided GUID

### ISSUE THREAD

**Issue by Whissi** [marcelklehr/floccus#726]
**Describe the bug**
Sync failed with error message **No bookmarks found for the provided GUID**

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Mozilla Firefox
 - Browser Version: 82.0.2
 - Floccus version: 4.4.6
 - Floccus sync method: nextcloud folders

**Server (please complete the following information):**
 - OS: Gentoo Linux
 - Nextcloud version: 19.0.4
 - Bookmarks app version: 3.4.9

Note: Please write down the actual version numbers instead of writing 'latest'.

**Debug log**
```
[...]
2020-11-09T21:27:29.153Z (nextcloud-folders)REMOVEFOLDER { folder: 
     + #undefined[Name] parentId: 323, hash: undefined
       - #Pd_arqNX7jE4[Link Name1](https://www.example.com/1) parentId: sAThGwwjLQ45
       - #WLSpEhYFRZPK[Link Name2](https://www.example.com/2) parentId: sAThGwwjLQ45
       - #irCCP_5xvhDJ[Link Name3](https://www.example.com/3) parentId: sAThGwwjLQ45
       - #LcURuRwCIFv3[Link Name4](https://www.example.com/4) parentId: sAThGwwjLQ45
       - #oLO4ws_78AuJ[Link Name5](https://www.example.com/5) parentId: sAThGwwjLQ45
       - #NTpBCDPZj4jc[Link Name6](https://www.example.com/6) parentId: sAThGwwjLQ45
       - #TSkyxAlxVQbA[Link Name7](https://www.example.com/7) parentId: sAThGwwjLQ45
       - #KqiUQiZNl8Ac[Link Name8](https://www.example.com/8) parentId: sAThGwwjLQ45
       - #l0yrU5l9zbQy[Link Name9](https://www.example.com/9) parentId: sAThGwwjLQ45
       - #iGrIOr3I8wAC[Link Name10](https://www.example.com/10) parentId: sAThGwwjLQ45 }
2020-11-09T21:27:29.153Z (nextcloud-folders)REMOVE { bookmark:   - #undefined[Link Name21](https://www.example.com/21) parentId: 323 }
[...]
2020-11-09T21:27:29.155Z (nextcloud-folders)REMOVE { bookmark:   - #undefined[Link Name67](https://www.example.com/67) parentId: 323 }
2020-11-09T21:27:29.161Z [Error: No bookmarks found for the provided GUID]
2020-11-09T21:27:29.161Z Syncing failed with No bookmarks found for the provided GUID
```

**Comment by rodgersan** [comment]
Same here on Fedora 33 with floccus 4.4.6 and firefox 82.0.3.
Can't provide server :(

**Comment by daddydaycare** [comment]
Yup, think this is the case and been around a long time now.

**Comment by Whissi** [comment]
Triggering sync from scratch for the affected account seems to have resolved the issue for me.

**Comment by rodgersan** [comment]
> Yup, think this is the case and been around a long time now. Now on Nextcloud 20.01 an current versions of firefox and floccus. Has existed now through many versions.. Keep testing and trying ...

Yep sorry, did not see bugs #690 and #694!

> Triggering sync from scratch for the affected account seems to have resolved the issue for me.

Did the trick, thanks!

**Comment by daddydaycare** [comment]
> > Yup, think this is the case and been around a long time now. Now on Nextcloud 20.01 an current versions of firefox and floccus. Has existed now through many versions.. Keep testing and trying ...
> 
> Yep sorry, did not see bugs #690 and #694!
> 
> > Triggering sync from scratch for the affected account seems to have resolved the issue for me.
> 
> Did the trick, thanks!

It always goes back to error for me.  If yours sticks or goes back to error please update. Love for this to work, keep coming back reinstalling and testing but same results.  Might try to use different ungoogled chromium and get rid of floccus to see if it works.

**Comment by gerroon** [comment]
This happens at random times for me. Sometimes I just add a  new bookmark and hit by this issue, sometimes it syncs fine. I am not sure what the issue is but it has been happening much more frequently in  last couple months.

**Comment by marcelklehr** [comment]
I'm sorry for the delay. I've just released v4.4.7 for firefox with a lot of new fixes, hopefully that will fix this as well. A few issues remain, but I'm confident I'll be able to tackle those soon.

**Comment by marcelklehr** [comment]
v4.4.8 is out now, which fixes this issue. Feel free to open a new issue if you continue to experience problems.
